### PR TITLE
feat: Add support to PUT items by URN

### LINKED ISSUE #489: Builder server not reporting the error correctly when updating an item/collection with an already used URN
## 🎉 Description

[Following the fix in the builder server](https://github.com/decentraland/builder/pull/1956) related to creating or updating a collection or an item that has an URN that is being already used, implement a mechanism to return a more detailed error that shows that the creation or update failed do to an URN being already used.

* [x] 🐛 This is a bug report.
* [ ] 📈 This is a feature request.

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #489

### CODE REVIEW

File: src/Item/Item.service.ts [hunk 1]
```diff
@@ -56,7 +56,15 @@ export class ItemService {
...
   ): Promise<FullItem> {
     let dbCollection: CollectionAttributes | undefined = undefined
-    const dbItem = await Item.findOne<ItemAttributes>(item.id)
+    const decodedItemURN =
+      !item.id && item.urn ? decodeThirdPartyItemURN(item.urn) : null
+    const dbItem = decodedItemURN
+      ? await Item.findByURNSuffix(
+          decodedItemURN.third_party_id,
+          decodedItemURN.item_urn_suffix
+        )
```
Comment (LautaroPetaccio) [inline src/Item/Item.service.ts:65]: If we don't fail when we're updating an item by URN and the item doesn't exist in the DB, we won't be able to tell the user that the editing process failed as it's always going to be creating a new item.
How does `upsertThirdPartyItem` behave when the item id and the item don't exist?

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]
There's a merge conflict to be solved, would you mind solving it before I can do the final review?

File: src/Item/Item.router.spec.ts [hunk 2]
```diff
@@ -56,6 +56,7 @@ import {
   ItemAttributes,
   ThirdPartyItemAttributes,
 } from './Item.types'
+import { omit } from 'decentraland-commons/dist/utils'
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/Item/Item.router.spec.ts]: done!

File: src/Item/Item.router.spec.ts [hunk 3]
```diff
@@ -1214,115 +1210,145 @@ describe('Item router', () => {
...
+                  data: {
+                    urn,
+                  },
+                  ok: false,
+                })
+              })
+          })
+        })
+
+        describe('and the item is being inserted by URN as param in the URI', () => {
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/Item/Item.router.spec.ts]: good catch! fixed!

File: src/Item/Item.router.ts [hunk 4]
```diff
@@ -326,15 +327,21 @@ export class ItemRouter extends Router {
...
+    let id, urn
+    if (isValidURN(idOrURN)) {
+      urn = idOrURN
+    } else {
+      id = idOrURN
+    }
     const itemJSON: FullItem = server.extractFromReq(req, 'item')
 
-    if (id !== itemJSON.id) {
+    if ((id && id !== itemJSON.id) || (urn && urn !== itemJSON.urn)) {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/Item/Item.router.ts:340]: The user won't be able to edit URNs in bullk.

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**